lib/LedgerSMB/Report/Unapproved/Batch_Overview.pm

```diff
@@ -160,49 +185,83 @@ Returns the inputs to display on header.
 
 sub header_lines {
     my ($self) = @_;
-    return [{name => 'batch_class',
-             text => $self->_locale->text('Batch Type')},
-            {name => 'reference',
-             text => $self->_locale->text('Reference')},
+    return [
+            {name => 'class_name',
+             text => $self->_locale->text('Transaction Type')},
+            {name => 'description',
+             text => $self->_locale->text('Description')},
             {name => 'amount_gt',
              text => $self->_locale->text('Amount Greater Than')},
             {name => 'amount_lt',
              text => $self->_locale->text('Amount Less Than')},
-            {name => 'locked',
-             text => $self->_locale->text('(Locked)')}, ]
+    ];
 }
 
 =head2 run_report()
 
-Runs the report, and assigns rows to $self->rows.
+Calls C<get_buttons> and C<get_rows> methods to query database for batches
+matching our filter properties, then populate C<rows> and C<buttons>
+properties.
 
 =cut
 
 sub run_report{
     my ($self) = @_;
-    $self->buttons([{
-                    name  => 'action',
-                    type  => 'submit',
-                    text  => $self->_locale->text('Post'),
-                    value => 'batch_approve',
-                    class => 'submit',
-                 },{
-                    name  => 'action',
-                    type  => 'submit',
-                    text  => $self->_locale->text('Delete'),
-                    value => 'batch_delete',
-                    class => 'submit',
-                 },{
-                    name  => 'action',
-                    type  => 'submit',
-                    text  => $self->_locale->text('Unlock'),
-                    value => 'batch_unlock',
-                    class => 'submit',
-                }]);
+    $self->get_buttons();
```

Comment:
Just wondering: shouldn't this be a functionality of the main `LedgerSMB::Report` class, as that allows rendering of input elements as well?

Reply:
Yes, `LedgerSMB::Report` expects that it's `set_buttons` method will be overrided to achieve this. Fixed in commit 6632dd4628b82e1f417051898d23dd8bb4499790